tests/rest/admin/test_room.py

```diff
@@ -972,6 +976,114 @@ def test_shutdown_room_block_peek(self) -> None:
         # Assert we can no longer peek into the room
         self._assert_peek(self.room_id, expect_code=403)
 
+    @unittest.override_config({"purge_retention_period": "1d"})
+    def test_purge_forgotten_room(self) -> None:
+        # Create a test room
+        room_id = self.helper.create_room_as(
+            self.admin_user,
+            tok=self.admin_user_tok,
+        )
+
+        self.helper.leave(room_id, user=self.admin_user, tok=self.admin_user_tok)
+        self.get_success(
+            self.room_member_handler.forget(
+                UserID.from_string(self.admin_user), room_id
+            )
+        )
+
+        # Test that room is not yet purged
+        with self.assertRaises(AssertionError):
+            self._is_purged(room_id)
+
+        self.reactor.advance(3600 * 24)
+
+        self._is_purged(room_id)
+
+    def test_resume_purge_room(self) -> None:
+        # Create a test room
+        room_id = self.helper.create_room_as(
+            self.admin_user,
+            tok=self.admin_user_tok,
+        )
+        self.helper.leave(room_id, user=self.admin_user, tok=self.admin_user_tok)
+
+        self.get_success(
+            self.store.upsert_room_to_purge(
+                room_id,
+                random_string(16),
+                DeleteStatus.STATUS_PURGING,
+            )
+        )
+
+        # Test that room is not yet purged
+        with self.assertRaises(AssertionError):
+            self._is_purged(room_id)
+
+        self.reactor.advance(3600 * 1)
```

Comment:
```suggestion
        # Advance one hour in the future past `PURGE_ROOMS_PERIOD` so that the automatic purging takes place and starts the scheduled purge
        self.reactor.advance(ONE_HOUR_IN_MS)
```

The `PURGE_ROOMS_PERIOD` variable here might need updating depending on how it's renamed, see https://github.com/matrix-org/synapse/pull/15488#discussion_r1199391705